router/default.go

```diff
@@ -426,6 +430,12 @@ func (r *router) advertiseEvents() error {
 					continue
 				}
 
+				// if we only advertise local routes delete and bail out
+				if local && advert.event.Route.Link != "local" {
```

Comment:
Makes me wonder whether we should skip adding the routes to the adverts map in the first place as soon as they're received.

Reply:
Yea good idea, done.